### PR TITLE
Fix a crash when multiple instances of Reader are opened after tapping on blogging prompts

### LINKED ISSUE #20843: Reader: 'Can't add self as subview' crash when closing the Reader
'Can't add self as subview' crash when closing the reader. Crash happens only sometimes and inconsistently. 

image

### Investigation

The inconsistent crash happens due to calling `super.viewDidAppear` in `viewDidDisappear` method

```
    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidAppear(animated)

        ReaderTracker.shared.stop(.readerPost)
    }
```

```
2023-06-12 14:34:45.191085+0300 Jetpack[93899:2704431] [ViewController] Calling -viewDidAppear: directly on a view controller is not supported, and may result in out-of-order callbacks and other inconsistent behavior. 0x15c942c00>

2023-06-12 14:34:46.149397+0300 Jetpack[93899:2704431] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Can't add self as subview'
```

**Related Stats issue:** https://github.com/wordpress-mobile/WordPress-iOS/issues/20849

### PR DESCRIPTION
**Pull request by staskus**
Fixes #20843

Reader topics view can be opened multiple times since `ReaderCoordinator.showTag()` performs an async network call to fetch a topic before presenting the controller. If users have a slower network connection, they can easily click multiple times, triggering multiple network calls, multiple completion blocks, and duplicating navigation in Reader, which eventually causes a crash.

### Solution:

Use already existing `controllerWithTagSlug` that makes the same `service.tagTopicForTag` async network request to fetch the topic after presenting Reader. It makes `showTag()` synchronous and prevents viewController from opening multiple items.

Other possible solutions:
- Disable the blogging prompt button while an async call is being made
- Show loading indicator and disable UI interaction while an async call is being made

I decided that these alternative solutions are less responsive and provide worse user experience.

**Note:** since similar issues are also logged in other places in the app (such as Stats), I will look for similar interactions in other areas of the app.

## To test:

1. Log in to a WP.com account
2. On the dashboard, tap the blogging prompts "View all responses" multiple times in quick succession
3. Reader topic view should open and eventually load a daily prompt topic
4. Navigate back
5. App shouldn't crash

## Regression Notes
1. Potential unintended areas of impact

Daily prompt navigation and prompt showing not working

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual tests

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by staskus** [comment]
@alpavanoglu I see you worked on blogging prompts, it would be great if you could take a look, maybe there was a reason you haven't used the `controllerWithTagSlug` as I started to use now! Thanks 🙇‍♂️!

File: WordPress/Classes/ViewRelated/System/Coordinators/ReaderCoordinator.swift [hunk 1]
```diff
@@ -58,29 +58,7 @@ class ReaderCoordinator: NSObject {
...
-        }
-    }
-
-    private func getTagTopic(tagSlug: String, completion: @escaping (Result<ReaderTagTopic, Error>) -> Void) {
-        let service = ReaderTopicService(coreDataStack: ContextManager.shared)
-        service.tagTopicForTag(withSlug: tagSlug,
-            success: { objectID in
-
-                guard let objectID = objectID,
-                    let topic = try? ContextManager.sharedInstance().mainContext.existingObject(with: objectID) as? ReaderTagTopic else {
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/System/Coordinators/ReaderCoordinator.swift:73]: not a blocker: This changes `ReaderTagTopic` to `ReaderAbstractTopic`. However, I don't see any of the additions that `ReaderTagTopic` adds are used here. I tested and both seem to work well.

**Review by guarani — APPROVED** [review]
I verified the issue can't be reproduced here anymore. Code changes look good, but I agree with checking with @alpavanoglu before merging 👍 🙇

File: WordPress/Classes/ViewRelated/System/Coordinators/ReaderCoordinator.swift (hunk 1 above)
Comment (staskus) [reply WordPress/Classes/ViewRelated/System/Coordinators/ReaderCoordinator.swift:73]: Good observation 👍 Yes, besides that I see no difference, especially given `ReaderTagTopic` is a child of `ReaderAbstractTopic`.

**Comment by alpavanoglu** [comment]
> @alpavanoglu I see you worked on blogging prompts, it would be great if you could take a look, maybe there was a reason you haven't used the controllerWithTagSlug as I started to use now! Thanks 🙇‍♂️!

Hey @staskus 
Apart from some reviews I actually have never worked on it. If you need input from someone who knows Blogging Prompts, @wargcm or @dvdchr would be the better people for that.

**Comment by staskus** [comment]
@wargcm Hey 👋 Do you mind taking a quick look if you worked on this part of the code before and you have time?

The main change here is fetching the topic for the tag after the `ReaderStreamViewController` is presented and not before, and using an already existing method to do that. The deleted `getTagTopic` in coordinator looks to be equivalent to `fetchTagTopic` in `ReaderStreamViewController`:

- Calling `ReaderTopicService(coreDataStack: ContextManager.shared).tagTopicForTag` and passing `tagSlug`
- Setting the result as `readerTopic` in `ReaderStreamViewController`

**Pull request merged**